src/model/element-base.ts

```diff
@@ -105,6 +106,25 @@ export interface ElementBaseInit extends ClassInit {
   readonly slots?: Slot[];
 }
 
+/**
+ * The element's runtime contents.
+ */
+export type ElementTemplate = {
+  /**
+   * HTML that is stamped out without data binding or other
+   * interpretation beyond normal HTML semantics.
+   */
+  kind: 'html',
+  contents: ASTNode,
+}|{
```

Comment:
This union either seems unnecessary or the content types are weird. For HTML-in-JS wouldn't we have a JS AST node, or a string? Or is this post-parsing by parse5? Is there a way to get back to the real source in case we want to modify it?

Reply:
This is post-parsing by parse5. In a follow-up PR I'm gonna add more affordances for working with this, including getting its parsed document for getting the original source, modifying, etc.